sql/core/src/main/scala/org/apache/spark/sql/execution/BaseScriptTransformationExec.scala

```diff
@@ -119,7 +119,7 @@ trait BaseScriptTransformationExec extends UnaryExecNode {
         val kvWriter = CatalystTypeConverters.createToCatalystConverter(StringType)
         prevLine: String =>
           new GenericInternalRow(
-            prevLine.split(outputRowFormat).slice(0, 2)
+            prevLine.split(outputRowFormat).slice(0, 2).padTo(2, null)
               .map(kvWriter))
```

Comment:
@maropu 
this question https://github.com/apache/spark/pull/29421/files#r469889057
 fixed by  this line. 

and will rase a new pr for https://issues.apache.org/jira/browse/SPARK-32667

Reply:
@maropu How about split this fix as a new one, since this fix is very necessary and no disputed

Reply:
If it's a trivial fix, IMO its okay to include it in this PR.